hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/thrift/HMSClient.java

```diff
@@ -0,0 +1,456 @@
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ *  Wrapper for Thrift HMS interface.
+ */
+public final class HMSClient implements AutoCloseable {
+  private static final Logger LOG = LoggerFactory.getLogger(HMSClient.class);
+  private static final String METASTORE_URI = "hive.metastore.uris";
+  private static final String CONFIG_DIR = "/etc/hive/conf";
+  private static final String HIVE_SITE = "hive-site.xml";
+  private static final String CORE_SITE = "core-site.xml";
+  private static final String PRINCIPAL_KEY = "hive.metastore.kerberos.principal";
+  private static final long SOCKET_TIMEOUT_MS = TimeUnit.SECONDS.toMillis(600);
+
+  private final String confDir;
+  private ThriftHiveMetastore.Iface client;
+  private TTransport transport;
+  private URI serverURI;
+
+  public URI getServerURI() {
+    return serverURI;
+  }
+
+  @Override
+  public String toString() {
+    return serverURI.toString();
+  }
+
+  /**
+   * Connect to HMS and return a client instance.
+   *
+   * @param uri Hive Metastore URI.
+   */
+  public HMSClient(@Nullable URI uri)
+      throws TException, IOException, InterruptedException, LoginException, URISyntaxException {
+    this(uri, CONFIG_DIR);
+  }
+
+  /**
+   * Connect to HMS and return a client instance.
+   *
+   * @param uri     Hive Metastore URI.
+   * @param confDir Directory with Hive configuration files - normally /etc/hive/conf
+   */
+  public HMSClient(@Nullable URI uri, @Nullable String confDir)
+      throws TException, IOException, InterruptedException, LoginException, URISyntaxException {
+    this.confDir = (confDir == null ? CONFIG_DIR : confDir);
+    getClient(uri);
+  }
+
+  /**
+   * Return a new connected client using the same configuration as the original client.
+   *
+   * @return New connected client instance
+   * @throws InterruptedException
+   * @throws URISyntaxException
+   * @throws TException
+   * @throws LoginException
+   * @throws IOException
+   */
+  public HMSClient doClone()
+      throws InterruptedException, URISyntaxException, TException, LoginException, IOException {
+    return new HMSClient(serverURI, confDir);
+  }
+
+  private void addResource(Configuration conf, @NotNull String r) throws MalformedURLException {
+    File f = new File(confDir + "/" + r);
+    if (f.exists() && !f.isDirectory()) {
+      LOG.debug("Adding configuration resource {}", r);
+      conf.addResource(f.toURI().toURL());
+    } else {
+      LOG.debug("Configuration {} does not exist", r);
+    }
+  }
+
+  /**
+   * Create a client to Hive Metastore.
+   * If principal is specified, create kerberised client.
+   *
+   * @param uri server uri
+   * @throws MetaException        if fails to login using kerberos credentials
+   * @throws IOException          if fails connecting to metastore
+   * @throws InterruptedException if interrupted during kerberos setup
+   */
+  private void getClient(@Nullable URI uri)
```

Comment:
can we fix method name. having "get" in the name, but method does not return anything.